src/setfit/modeling.py

```diff
@@ -243,13 +245,29 @@ def fit(
             self.model_head.fit(embeddings, y_train)
 
     def _prepare_dataloader(
-        self, x_train: List[str], y_train: List[int], batch_size: int, shuffle: bool = True
+        self,
+        x_train: List[str],
+        y_train: List[int],
+        batch_size: int,
+        max_length: Optional[int] = None,
+        shuffle: bool = True,
     ) -> DataLoader:
+        max_acceptable_length = self.model_body.get_max_seq_length()
+        max_length = max_length or max_acceptable_length
+        if max_length > max_acceptable_length:
+            warnings.warn(
+                (
+                    f"The specified `max_length`: {max_length} is greater than the maximum length of the current model body: {max_acceptable_length}. "
+                    f"Change `max_length` to {max_acceptable_length}."
```

Comment:
Since we're overwriting the user input, should we indicate that explicitly? Alternatively, we can raise an error if you think that's a better approach
```suggestion
                    f"Using {max_acceptable_length} instead."
```